yoti_python_sdk/activity_details.py

```diff
@@ -42,33 +47,48 @@ def try_parse_selfie_field(self, field):
             field.content_type
         )
 
-    def try_parse_age_verified_field(self, field):
+    def try_parse_age_verified_field(self, field, anchors):
         if field is not None:
             is_age_verified = Protobuf().value_based_on_content_type(
                 field.value,
                 field.content_type
             )
             if is_age_verified == 'true':
                 self.user_profile['is_age_verified'] = True
+                self.profile['is_age_verified'] = attribute.attribute(is_age_verified, True, anchors)
                 return
             if is_age_verified == 'false':
                 self.user_profile['is_age_verified'] = False
+                self.profile['is_age_verified'] = attribute.attribute(is_age_verified, False, anchors)
                 return
 
         raise TypeError("age_verified_field unable to be parsed")
 
-    def try_convert_structured_postal_address_to_dict(self, field):
+    def try_convert_structured_postal_address_to_dict(self, field, anchors):
         decoder = json.JSONDecoder(object_pairs_hook=collections.OrderedDict)
         self.user_profile[config.ATTRIBUTE_STRUCTURED_POSTAL_ADDRESS] = decoder.decode(field.value)
-
-    def set_address_to_be_formatted_address_if_null(self):
-        if 'postal_address' not in self.user_profile and 'structured_postal_address' in self.user_profile:
-            if 'formatted_address' in self.user_profile['structured_postal_address']:
-                self.user_profile['postal_address'] = self.user_profile['structured_postal_address'][
-                    'formatted_address']
+        self.profile[config.ATTRIBUTE_STRUCTURED_POSTAL_ADDRESS] = attribute.attribute(
+            config.ATTRIBUTE_STRUCTURED_POSTAL_ADDRESS,
+            decoder.decode(field.value),
+            anchors)
+
+    def set_address_to_be_formatted_address_if_null(self, anchors):
+        if config.ATTRIBUTE_POSTAL_ADDRESS not in self.user_profile and config.ATTRIBUTE_STRUCTURED_POSTAL_ADDRESS in self.user_profile:
+            if config.KEY_FORMATTED_ADDRESS in self.user_profile[config.ATTRIBUTE_STRUCTURED_POSTAL_ADDRESS]:
+                self.user_profile[config.ATTRIBUTE_POSTAL_ADDRESS] = self.user_profile[config.ATTRIBUTE_STRUCTURED_POSTAL_ADDRESS][
+                    config.KEY_FORMATTED_ADDRESS]
+
+        if config.ATTRIBUTE_POSTAL_ADDRESS not in self.profile and config.ATTRIBUTE_STRUCTURED_POSTAL_ADDRESS in self.profile:
```

Comment:
My understanding is that you are managing backward compatibility with user_profile here and that's why we twice the same condition with two objects. I'm not sure whether it's worth refactoring it, but a comment would help, because it is a bit tough to read.

Reply:
I did think about refactoring, but thought I'd leave it how it would be best once the .user_profile object and associated methods were removed. 

You're right it's tough to read though, I'll add a comment